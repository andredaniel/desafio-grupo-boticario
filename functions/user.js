const axios = require('axios')

const randomStatus = () => {
  const key = Math.floor(Math.random() * 3)

  const statuses = ['APPROVED', 'DISAPPROVED', 'IN_VALIDATION']

  return statuses[key]
}

exports.handler = async function (event, context) {
  const baseUrl = 'https://5fffb468cb21e10017af7e99.mockapi.io/api'

  if (event.httpMethod === 'GET') {
    const { data } = await axios.get(
      `${baseUrl}/orders/?user_id=${event.queryStringParameters.user_id}`
    )

    return {
      statusCode: 200,
      body: JSON.stringify(data),
    }
  }

  if (event.httpMethod === 'POST') {
    const newOrder = JSON.parse(event.body)

    newOrder.cashbackAppliedPercentage = Math.ceil(Math.random() * 20)
    newOrder.cashbackReceivedValue =
      (newOrder.value / 100) * newOrder.cashbackAppliedPercentage
    newOrder.status = randomStatus()

    const { data } = await axios.post(`${baseUrl}/orders`, newOrder)

    return {
      statusCode: 200,
      body: JSON.stringify(data),
    }
  }
}
