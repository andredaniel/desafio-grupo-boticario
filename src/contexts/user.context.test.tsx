import { render } from '@testing-library/react'
import React from 'react'
import { UserContext, UserProvider, useUser } from './user.context'

describe('User context', () => {
  test('UserContext', () => {
    expect(UserContext).toBeDefined()
  })

  it('should render UserProvider component as expected', () => {
    const { container } = render(<UserProvider />)
    expect(container).toMatchSnapshot()
  })

  test('useUser', () => {
    expect(useUser).toBeDefined()
  })
})
