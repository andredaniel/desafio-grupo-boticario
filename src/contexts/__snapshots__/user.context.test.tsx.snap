// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User context should render UserProvider component as expected 1`] = `<div />`;
